hub/core/chunk_engine.py

```diff
@@ -188,6 +188,8 @@ def _append_bytes(self, buffer: memoryview, shape: Tuple[int], dtype: np.dtype):
             dtype (np.dtype): Data type for the sample that `buffer` represents.
         """
 
+        self.cache.check_readonly()
```

Comment:
without this, tensor meta length was being incremented even when data uploads would fail. the test i added covers this

Reply:
I'm not sure how this addresses the issue. Could you elaborate?

Reply:
it was really late i think i mixed up concepts here lol

Reply:
the test covers it though and it passes so all good